spec/support/ansible_shared/automation_manager/refresher.rb

```diff
@@ -113,6 +113,11 @@ def assert_counts
   def assert_credentials
     expect(expected_configuration_script.authentications.count).to eq(3)
 
+    # vault_credential
+    vault_credential = Authentication.all.find_by(:type => manager_class::VaultCredential)
+    expect(vault_credential.options.keys).to match_array([:vault_password])
+    expect(vault_credential.name).to eq("Demo Creds 2")
```

Comment:
@jameswnl - I'm not sure I'm seeing a reason to assert this name as it doesn't help determine the `vault_credential` Type. 

Do you think there would be value in asserting that the vault credential is really a `machine_credential` that includes a kind of `ssh` and a non empty `vault_password`?

Reply:
There's really nothing else to check as `vault_credential` really has only 1 piece of info interesting,  i.e. the `vault_password` and it is inside the hash stored in the `options` column.

In the parsing, we already make a decision to classify this as a `vault_credential` and not related to `machine_credential`.

I've added a check to make sure the `options.vault_password` not empty.